app/experimenter/openidc/tests/test_middleware.py

```diff
@@ -69,3 +69,45 @@ def test_user_created_with_correct_email_from_header(self):
 
         self.assertEqual(request.user.email, user_email)
         self.assertFalse(request.user.is_staff)
+
+    def test_dev_user_is_super_staff_user_when_debug_true(self):
+        dev_email = "dev@example.com"
+
+        self.assertEqual(dev_email, settings.DEV_USER_EMAIL)
+
+        request = mock.Mock()
+        request.META = {settings.OPENIDC_EMAIL_HEADER: dev_email}
+
+        User = get_user_model()
+
+        self.assertEqual(User.objects.all().count(), 0)
+
+        with self.settings(OPENIDC_AUTH_WHITELIST=[], DEBUG=True):
+            response = self.middleware(request)
+
+        self.assertEqual(response, self.response)
+        self.assertEqual(User.objects.all().count(), 1)
+
+        self.assertEqual(request.user.email, dev_email)
+        self.assertTrue(request.user.is_staff)
+        self.assertTrue(request.user.is_superuser)
+
+    def test_dev_user_is_not_super_staff_user_when_debug_false(self):
+        dev_email = settings.DEV_USER_EMAIL
+
+        request = mock.Mock()
+        request.META = {settings.OPENIDC_EMAIL_HEADER: dev_email}
+
+        User = get_user_model()
+
+        self.assertEqual(User.objects.all().count(), 0)
+
+        with self.settings(OPENIDC_AUTH_WHITELIST=[]):
```

Comment:
Don't depend on whatever the DEBUG setting is set to in the test environment, explicitly set it here, as well as the email like I mentioned for the other test.